### ISSUE TITLE
Codeintel go to definition finds nothing

### ISSUE THREAD

**Issue by tuckdesign** [Komodo/KomodoEdit#3568]
### Short Summary
When you try to 'Go to definition' on any bigger PHP project, 'go to anywhere' shows 'could not find definition'. This works on really tiny peaces of codes...

### Steps to Reproduce
Open some big PHP project and right click on some function name where function definition is in another file in project. Choose 'Go To Definition'. 

### Expected results
File with function should be opened and cursor put on function definition.

### Actual results
Nothing happens except message in 'Go to Anything' field that says 'Could not find definition'

### Platform Information

Komodo IDE
11.1 build 91033
Mac OS 10.13.5

### Additional Information
Tried clearing codeintel DB under help/troubleshooting. Tried deleting db from filesystem in 10.0 folder and deleting komodo ide 11 folder and then restarting. Nothing helped.

**Comment by mitchell-as** [comment]
Hi, unfortunately I'm not able to reproduce without something more concrete. I tested on a blank Wordpress project and goto definition seems to work fine. (Granted it was not exhaustive testing.) I know it may be difficult, but do you have a sample that you could share that demonstrates the problem?

**Comment by tuckdesign** [comment]
@mitchell-as If you have bitbucket account I can share project with you.

**Comment by goozak** [comment]
I'm experiencing the same problem.  Don't think I'll have the time to make a shareable sample, though...

**Comment by tuckdesign** [comment]
I have code.... but @mitchell-as didn't send me email or bitbucket account yet so I can share it with him

**Comment by Naatan** [comment]
@tuckdesign please share it with "Naatan", I'll make sure it gets shared internally. Mitchell is out until Thursday because of some smart 4th of July planning on his part ;)

**Comment by tuckdesign** [comment]
@Naatan shared. For example open custom/Extension/modules/Schedulers/Ext/ScheduledTasks/email_reminder.php and try 'Go to definition' on eg line 130: '::getInstance()' or basically any other place...

**Comment by Naatan** [comment]
Hmm I didn't receive anything on bitbucket. Are you sure you shared it with "Naatan" ?

**Comment by tuckdesign** [comment]
Sorry, I missed notification from github. @Naatan I added you as user, maybe notification was not sent. Check this: https://bitbucket.org/tuckdesign/ablux

**Comment by Naatan** [comment]
Thanks @tuckdesign, got it.

For us devs, you can access the files here: https://activestate.slack.com/archives/C03NU4MN5/p1531752125000607

**Comment by mitchell-as** [comment]
After investigation I have identified the problem. The PHP auto-importer does not auto-import everything the project, but only things in a sub-directory of the current file. The following minimal test-case fails:

```
    def test_superclass_goto_definition(self):
        self.assertDefinitionIs("""\
            <?php
            class Foo {}

            class Bar extends <|>Foo {}
        """, line=2)

        self.assertDefinitionIs({
            os.path.join("foo", "bar", "baz.php"): """\
                <?php
                class Baz {
                    public function Baz() {}
                }
            """,
            os.path.join("bar", "baz", "quux.php"): """\
                <?php

                require_once('foo/bar/baz.php');
                class Quux extends <|>Baz {}
            """
        }, file=os.path.join("foo", "bar", "baz.php"), line=2)
```
If `quux.php` was in the top-level project directory, the test would pass.

**Comment by mitchell-as** [comment]
@Naatan feel free to prioritize this for 11.1.1 if you deem it necessary. I think this was introduced in 11.0.x though, not 11.1.

**Comment by Naatan** [comment]
I'll prioritise but on the condition that this should ONLY land in 11.1.1 if it has negligible performance impact.

**Comment by mitchell-as** [comment]
After further review I have determined that I did NOT identify the problem after all. I was looking in the wrong place. CodeIntel should be auto-importing from everything you've declared in Project > Project Preferences > Languages > PHP > PHP Directories. Now, large PHP projects will cause trouble time-wise. The linked project for example always timed out goto-definition requests when I specified the root directory for auto-importing symbols from. For a smaller-scale test, I specified only the "models/users" directory for auto-importing, opened up the root-directory's "cron.php" file, and ran goto-definition for "User". After a while Komodo brought me to the class definition in "models/users/User.php".

The conclusion is CodeIntel does work, but it's prone to timeout for really large projects like yours. The action item is to ignore timeouts for goto-definition.
